tests/test_config.py

```diff
@@ -35,3 +41,28 @@ def test_configure_sentry_env_variable_is_dsn(monkeypatch):
     monkeypatch.setenv("SENTRY_DSN", "https://1234567890@00000.ingest.sentry.io/123456")
     result = configure_sentry()
     assert result == "Sentry DSN found, exceptions will be sent to Sentry with env=test"
+
+
+def test_sentry_scope_variables_removed_from_sent_event(
+    mocker,
+    sensitive_scope_variable,
+):
+    not_secret_value = "Everyone can see this exception message."
+    secret_value = "very-secret-abc123"
+
+    spy_scrubber = mocker.spy(config, "_remove_sensitive_scope_variables")
+    try:
+        raise Exception(not_secret_value)
+    except Exception as exc:
+        sentry_sdk.capture_exception(exc)
+
+    original_event = json.dumps(spy_scrubber.call_args)
+    scrubbed_event = json.dumps(spy_scrubber.spy_return)
+
+    # not secret value present before and after scrubbing
+    assert not_secret_value in original_event
+    assert not_secret_value in scrubbed_event
```

Comment:
Well-designed test, this really demonstrates the value of `pytest-mock` for tricky situations when a dependency like `sentry_sdk` is involved

Reply:
I felt the same.  The actual scrubbing of data was easy, but testing was suprisingly difficult.  When the function is registered for `before_send`, it becomes unavailable to mock.  And, it's difficult to prevent this from happening given the structure of the CLI.

But yeah, `pytest-mock`'s "spy" feature is amazing.  It's what I always wanted, and finally took the time to do some digging.  My two cents is that it should be part of the stdlib for `pytest`.